pkg/client/rootd/dns/server.go

```diff
@@ -666,6 +728,28 @@ func (s *Server) resolveQuery(q *dns.Question, dv *cacheEntry) (dnsproxy.RRs, in
 		close(dv.wait)
 	}()
 
+	// Returns a CNAME pointing to the mapping when there is a hit.
+	if mappingAlias := s.ResolveMappingAlias(q.Name); mappingAlias != nil {
+		dv.answer = dnsproxy.RRs{&dns.CNAME{
+			Hdr:    dns.RR_Header{Name: q.Name, Rrtype: dns.TypeCNAME, Class: dns.ClassINET, Ttl: dnsTTL},
+			Target: *mappingAlias,
+		}}
+		// On Windows, or in a Linux container, just returning the cname isn't enough, and the DNS resolver won't try
+		// to get the record with a subsequent request, so we need to resolve the record until we get a better solution.
+		if runtime.GOOS == "windows" || proc.RunningInContainer() {
```

Comment:
I ended up having the same problem in a container than on Windows. Could it be because in both cases, we intercept the DNS queries to redirect them to our local resolver, so it doesn't work well with Cnames ?